src/main/java/net/seesharpsoft/intellij/plugins/csv/editor/table/CsvTableEditor.java

```diff
@@ -85,8 +86,12 @@ public final void updateTableComponentData(Object[][] values) {
         }
     }
 
+    public int getRowLineHeight() {
+        return Math.round((float)(getFileEditorState().getFontSize() * LINE_HEIGHT_RATIO));
+    }
+
     public int getPreferredRowHeight() {
-        return ROW_LINE_HEIGHT * getFileEditorState().getRowLines();
+        return getRowLineHeight() * getFileEditorState().getRowLines();
```

Comment:
I noticed that this calculation is quite inaccurate.

Some background: The used `MultiLineCellRenderer` (extending `JTextArea`) comes with a [getPreferredSize](https://docs.oracle.com/javase/8/docs/api/javax/swing/JTextArea.html#getPreferredSize--) method which makes use of `FontMetric` and takes the border and its insets into account. After some digging, the actual height of a row boils down to

```
totalHeight = getFileEditorState().getRowLines() * getTable().getFontMetrics(getTable().getFont()).getHeight() + 3;
// Note: 3 = topBorder inset + bottomBorder inset + border itself
```

Please note that this is a Swing specific implementation (needs to done in CsvTableEditorSwing).

So `getPreferredRowHeight` should become abstract, and I suggest for the overriding method in CsvTableEditorSwing:

```
private static final int TOTAL_CELL_HEIGHT_SPACING = 3;

@Override
public int getPreferredRowHeight() {
    if (getFileEditorState().getRowLines() == 0) {
        return 0;
    }
    return getFileEditorState().getRowLines() * getTable().getFontMetrics(getTable().getFont()).getHeight() + TOTAL_CELL_HEIGHT_SPACING ;
}
```

`getRowLineHeight` would become obsolete

Reply:
Would it be better , if the spacing height is not a constant (TOTAL_CELL_HEIGHT_SPACING) , but proportional to the font height?